libsolidity/experimental/analysis/TypeInference.cpp

```diff
@@ -216,8 +216,7 @@ bool TypeInference::visit(TypeClassDefinition const& _typeClassDefinition)
 		subNode->accept(*this);
 		auto const* functionDefinition = dynamic_cast<FunctionDefinition const*>(subNode.get());
 		solAssert(functionDefinition);
-		// TODO: need polymorphicInstance?
-		auto functionType = polymorphicInstance(typeAnnotation(*functionDefinition));
+		auto functionType = typeAnnotation(*functionDefinition);
```

Comment:
I still need to confirm that removing this does not break things and makes sense.